third_party/move/move-vm/types/src/value_serde.rs

```diff
@@ -89,25 +102,29 @@ pub fn deserialize_and_allow_delayed_values(
     bytes: &[u8],
     layout: &MoveTypeLayout,
 ) -> Option<Value> {
-    let native_deserializer = RelaxedCustomSerDe;
+    let native_deserializer = RelaxedCustomSerDe { delayed_fields_count: RefCell::new(0) };
     let seed = DeserializationSeed {
         custom_deserializer: Some(&native_deserializer),
         layout,
     };
-    bcs::from_bytes_seed(seed, bytes).ok()
+    bcs::from_bytes_seed(seed, bytes)
+        .ok()
+        .filter(|_| native_deserializer.delayed_fields_count.into_inner() <= MAX_DELAYED_FIELDS_PER_RESOURCE)
```

Comment:
Why not fail when incrementing? Though processing extra aggregators is not really a problem, so should be fine :)

Reply:
because we later want to extend this to charging, so correct value will be needed :)